### PR TITLE
LPS-195935 Add infopanel for LayoutPageTemplateEntries and LayoutPageTemplateCollections

### PR DESCRIPTION
**Pull request by YurenaC**
[Story](https://liferay.atlassian.net/browse/LPS-195935)

Motivation
=======
The infopanel is currently now available for folders or DPTs created in Design > Page Templates > Display Page Templates.

Proposed Solution
========
Add the infopanel.

Steps to Verify
========
1. Go to Design > Page Templates > Display Page Templates,  create several folders (nested ones) and DPTs.
2. Ensure the infopanel show the right information in the following way:
3. When nothing is selected, the infopanel show the current folder info
4. When a DPT is selected, the infopanel shows the information as described in the [figma](https://www.figma.com/file/ydHldCxSGZuPe1tuHu9jXe/LPS-189856---Usability-improvements-for-objects-page-builder-integration?type=design&node-id=1-453&mode=design&t=dcDO8RIFKzjTGYaU-0) attached to the original story.
5.  When a folder is selected, the infopanel shows the information as described in the [figma](https://www.figma.com/file/ydHldCxSGZuPe1tuHu9jXe/LPS-189856---Usability-improvements-for-objects-page-builder-integration?type=design&node-id=1-453&mode=design&t=dcDO8RIFKzjTGYaU-0) attached to the original story.
6. When multiple items are selected, the info panel inly show the number of elements that have been selected.

Commits
========

- LPS-195935 Create webkeys to allow selection of LayoutPageTemplatesEntries and LayoutPageTemplatesCollections to apply different actions
- LPS-195935 Add method to retrieve actual LayoutPageTemplateCollection
- LPS-195935 Override getInfoPanelId method
- LPS-195935 Override getInputName methods to know selected LayoutPageTemplateEntries and LayoutPageTemplateCollections
- LPS-195935 Add infopanel option to LayoutPageTemplates view
- LPS-195935 Display infopanel for LayoutPageDisplayCollections
- LPS-195935 Showing current layoutPageTemplateCollection info in infopanel when nothing is selected
- LPS-195935 Delete layoutPageTemplateCollection and layoutPageTemplateEntries from Display Page Templates tab
- LPS-195935 Display infopanel for LayoutPageTemplateEntries

### CODE REVIEW

**Comment by liferay-echo** [comment]
Thank you for submitting this PR. 👏

Make sure it follows the [Echo Pull Request Template](https://gist.github.com/liferay-echo/5a4455285ff1800f9da146816d85e67d#file-pull-request-template-md) if you haven't already done so. 

Learn more about it [here](https://liferay.atlassian.net/wiki/spaces/ENGECHO/pages/2027455014/Echo+Pull+Request+Etiquette).

**Comment by YurenaC** [comment]
ci:test:sf

File: modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/DeleteLayoutPageTemplateEntriesAndLayoutPageTemplateCollectionsMVCActionCommand.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+@Component(
+	property = {
+		"javax.portlet.name=" + LayoutPageTemplateAdminPortletKeys.LAYOUT_PAGE_TEMPLATES,
+		"mvc.command.name=/layout_page_template_admin/delete_layout_page_template_entries_and_layout_page_template_collections"
+	},
+	service = MVCActionCommand.class
+)
+public class
+	DeleteLayoutPageTemplateEntriesAndLayoutPageTemplateCollectionsMVCActionCommand
+		extends BaseMVCActionCommand {
```
Comment (jkappler) [inline modules/apps/layout/layout-page-template-admin-web/src/main/java/com/liferay/layout/page/template/admin/web/internal/portlet/action/DeleteLayoutPageTemplateEntriesAndLayoutPageTemplateCollectionsMVCActionCommand.java:33]: @YurenaC, can we make this a transactional command?

File: modules/apps/layout/layout-page-template-admin-web/src/main/resources/META-INF/resources/info_panel.jsp [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+ */
+--%>
+
+<%@ include file="/init.jsp" %>
+
+<%
+List<LayoutPageTemplateCollection> layoutPageTemplateCollections = (List<LayoutPageTemplateCollection>)request.getAttribute(LayoutPageTemplateAdminWebKeys.LAYOUT_PAGE_TEMPLATE_COLLECTIONS);
+List<LayoutPageTemplateEntry> layoutPageTemplateEntries = (List<LayoutPageTemplateEntry>)request.getAttribute(LayoutPageTemplateAdminWebKeys.LAYOUT_PAGE_TEMPLATE_ENTRIES);
+
+if (ListUtil.isEmpty(layoutPageTemplateCollections) && ListUtil.isEmpty(layoutPageTemplateEntries)) {
```
Comment (jkappler) [inline modules/apps/layout/layout-page-template-admin-web/src/main/resources/META-INF/resources/info_panel.jsp:14]: @YurenaC, can we move all logic to a display context? That way it's easier to debug in the future

File: modules/apps/layout/layout-page-template-admin-web/src/main/resources/META-INF/resources/info_panel.jsp [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+			<%
+			long layoutPageTemplateCollectionId = ParamUtil.getLong(request, "layoutPageTemplateCollectionId");
+
+			LayoutPageTemplateCollection layoutPageTemplateCollection = LayoutPageTemplateCollectionLocalServiceUtil.fetchLayoutPageTemplateCollection(layoutPageTemplateCollectionId);
+
+			List<String> paths = new ArrayList<>();
+			%>
+
+			<c:if test="<%= layoutPageTemplateCollection == null %>">
+				Null again
```
Comment (jkappler) [inline modules/apps/layout/layout-page-template-admin-web/src/main/resources/META-INF/resources/info_panel.jsp:69]: Null again? Not null? 🤔

File: modules/apps/layout/layout-page-template-admin-web/src/main/resources/META-INF/resources/info_panel.jsp [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+			<c:if test="<%= layoutPageTemplateCollection == null %>">
+				<p class="sidebar-dd text-secondary">
+					<%= LayoutPageTemplateEntryServiceUtil.getLayoutPageCollectionsAndLayoutPageTemplateEntriesCount(scopeGroupId, LayoutPageTemplateConstants.PARENT_LAYOUT_PAGE_TEMPLATE_COLLECTION_ID_DEFAULT, LayoutPageTemplateEntryTypeConstants.TYPE_DISPLAY_PAGE) %>
+				</p>
+			</c:if>
+
+			<c:if test="<%= layoutPageTemplateCollection != null %>">
+				<p class="sidebar-dd text-secondary">
+					<%= LayoutPageTemplateEntryServiceUtil.getLayoutPageCollectionsAndLayoutPageTemplateEntriesCount(layoutPageTemplateCollection.getGroupId(), layoutPageTemplateCollection.getLayoutPageTemplateCollectionId(), layoutPageTemplateCollection.getType()) %>
```
Comment (jkappler) [inline modules/apps/layout/layout-page-template-admin-web/src/main/resources/META-INF/resources/info_panel.jsp:151]: @YurenaC, as I mentioned, lets try to have all of this in the display context

**Comment by jkappler** [comment]
@YurenaC, other than fixing the conflicts, let's try to move what we can to display context, it will help a lot in the future ❤️

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by jkappler** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
@YurenaC, we have to hide info panel under a FF

**Comment by jkappler** [comment]
ci:stop

**Comment by YurenaC** [comment]
ci:test:sf

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:stop

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by jkappler** [comment]
ci:forward

**Comment by jkappler** [comment]
Baseline issue is not related to the PR. There are no changes to layout-api

**Comment by ealonso** [comment]
ci:forward:force

**Pull request closed without merging**